File: C#/Queue.cs
using System;

public class Node
{
    Node link;
    int data;
    public Node(node, data)
    {
        link = Node;
        data = data;
    }
}

public class Queue
{
    Node front;
    Node back;

    public Queue()
    {
        front = null;
        back = null;
    }

    public void enqueue(int data)
    {
        if (back == null && front == null)
        {
            back = new Node(null, data);
            front = back;
        }
        else
        {
            Node temp = new Node(back, data);
            back = temp;
        }
    }

    public int dequeue()
    {
        if (front == null)
        {
            Console.WriteLine("Queue Empty");
        }
        return front.data;
    }

    public int peek()
    {
        if (front == null) { Console.WriteLine("Queue Empty"); }
        return front.data;
    }

    public bool isEmpty()
    {
        if (back == null && front == null) { return true; }
        return false;
    }

    public int size()
    {
        Node temp = back;
        int counter = 0;
        while (temp != null)
        {
            counter++;
            temp = temp.link;
        }
        return counter;
    }

    public void printList()
    {
        Node temp = back;
        while (temp != null)
        {
            Console.WriteLine(temp.data);
            temp = temp.link;
        }
    }

    public static void Main(string[] args)
    {
        Queue queue = new Queue();

        Console.WriteLine(queue.isEmpty());
        queue.enqueue(1);
        queue.enqueue(3);
        queue.enqueue(5);
        queue.enqueue(7);
        queue.enqueue(9);
        Console.WriteLine(queue.isEmpty());
        queue.printList();
        queue.pop();
        queue.printList();
    }
}

File: Go/Queue.go
package main
import "fmt"

type Node struct {
	data int
	link *Node
}

var front *Node
var back *Node

func enqueue(data int){
	temp := &Node{data, (&Node{})}
	if front == nil {
		back = temp
		front = temp
	} else {
		back.link = temp
		back = temp
	}
}

func dequeue() int {
	if (front == nil){
		fmt.Println("Queue Empty!")
	}
	data := front.data
	front = front.link
	return data
}

func peek() int {
	if front == nil {
		fmt.Println("Queue Empty!")
	}
	return front.data
}

func isEmpty() bool {
	if (front == nil){
		return true
	}
	return false
}

func size() int {
	temp := front
	var counter int = 0
	for {
		if temp == nil {
			break
		}
		counter++
		temp = temp.link
	}
	return counter
}

func printList(){
	temp := front
	for {
		if temp == nil {
			break
		}
		fmt.Println(temp.data)
		temp = temp.link
	}
}

func main(){
	fmt.Println(isEmpty())

	enqueue(1)
	enqueue(3)
	enqueue(5)
	enqueue(7)
	enqueue(9)

	printList()

	fmt.Println(isEmpty())
	fmt.Println(size())
	fmt.Println(peek())
	fmt.Println(dequeue())

	printList()
}
